lib/modules/manager/bazel/parser.ts

```diff
@@ -0,0 +1,204 @@
+      ctx.result.meta.push({
+        path: [...ctx.currentMetaPath],
+        data: { offset: offset, length: value.length },
+      });
+      return ctx;
+    }),
+    // array of strings case
+    q.tree({
+      type: 'wrapped-tree',
+      maxDepth: 1,
+      preHandler: (ctx) => {
+        ctx.currentArray = [];
+        return ctx;
+      },
+      search: q.str<Ctx>((ctx, { value, offset }) => {
+        ctx.result.meta.push({
+          path: [...ctx.currentMetaPath, ctx.currentArray!.length],
+          data: { offset: offset, length: value.length },
+        });
+        ctx.currentArray?.push(value);
+        return ctx;
+      }),
+      postHandler: (ctx) => {
+        ctx.currentTarget[ctx.currentAttrKey!] = ctx.currentArray;
+        ctx.currentArray = [];
+        return ctx;
+      },
+    })
+  );
+
+/**
+ * Matches rule signature, i.e. content of `git_repository(...)`
+ *
+ * @param search something to match inside parens
+ */
+function ruleCall(search: q.QueryBuilder<Ctx>): q.QueryBuilder<Ctx> {
+  return q.tree({
+    type: 'wrapped-tree',
+    maxDepth: 1,
+    preHandler: (ctx) => {
+      ctx.currentMetaPath.push(ctx.ruleIndex);
+      return ctx;
+    },
+    search,
+    postHandler: (ctx, tree) => {
+      // TODO: remove it (#9667)
+      if (is.number(ctx.ruleStartOffset) && tree.type === 'wrapped-tree') {
+        const { children, endsWith } = tree;
+        const lastElem = children[children.length - 1];
+        if (lastElem.type === '_end') {
+          const ruleEndOffset = lastElem.offset + endsWith.value.length;
+          const offset = ctx.ruleStartOffset;
+          const length = ruleEndOffset - ctx.ruleStartOffset;
+          ctx.result.meta.push({
+            path: [ctx.ruleIndex],
+            data: { offset, length },
+          });
+          delete ctx.ruleStartOffset;
+        }
+      }
+
+      if (isTarget(ctx.currentTarget)) {
+        ctx.result.targets.push(ctx.currentTarget);
+      }
+      ctx.currentTarget = {};
+      ctx.ruleIndex += 1;
+      ctx.currentMetaPath.pop();
+
+      return ctx;
+    },
+  });
+}
+
+/**
+ * Matches regular rules:
+ * - `git_repository(...)`
+ * - `go_repository(...)`
+ */
+const regularRule = ruleSym.join(ruleCall(kwParams));
+
+const maybeFirstArg = q.begin<Ctx>().join(ruleSym).op(',');
+
+/**
+ * Matches "maybe"-form rules:
+ * - `maybe(git_repository, ...)`
+ * - `maybe(go_repository, ...)`
+ */
+const maybeRule = q
+  .sym<Ctx>(
+    'maybe',
+    // TODO: remove it (#9667)
+    (ctx, { offset }) => {
+      if (!is.number(ctx.ruleStartOffset)) {
+        ctx.ruleStartOffset = offset;
+      }
+      return ctx;
+    }
+  )
+  .join(ruleCall(q.alt(maybeFirstArg, kwParams)));
+
+const rule = q.alt<Ctx>(maybeRule, regularRule);
+
+const query = q.tree<Ctx>({
+  type: 'root-tree',
+  maxDepth: 16,
+  search: rule,
+});
+
+async function getCacheKey(input: string): Promise<string> {
+  const hash = await hasha.async(input, { algorithm: 'sha512' });
```

Comment:
```suggestion
  const hash = await hasha(input);
```
i think async is too much overhead, as it will spawn a new process ( i think I found that on npm package readme)

Reply:
Okay, reverted this one